dap_prinz_green_jobs/pipeline/ojo_application/extract_green_measures.py

```diff
@@ -0,0 +1,147 @@
+--------------
+A pipeline that extracts green measures on a OJO sample
+    as defined in dap_prinz_green_jobs/getters/data_getters/ojo.py
+
+It also saves the extracted green measures to s3.
+
+To run the script including with time tracking:
+    time python dap_prinz_green_jobs/pipeline/ojo_application/extract_green_measures.py
+
+In production, this script should take about ~15 minutes to run now
+"""
+from dap_prinz_green_jobs.pipeline.green_measures.green_measures import GreenMeasures
+import dap_prinz_green_jobs.pipeline.green_measures.skills.skill_measures_utils as sm
+
+from dap_prinz_green_jobs.getters.ojo_getters import (
+    get_ojo_job_title_sample,
+    #    get_extracted_skill_embeddings,
+    #    get_ojo_skills_sample,
+    get_mapped_green_skills,
+)
+
+# from dap_prinz_green_jobs.getters.skill_getters import (
+#     get_green_skills_taxonomy,
+#     get_green_skills_taxonomy_embeddings,
+
+# )
+
+from dap_prinz_green_jobs import logger
+from dap_prinz_green_jobs.getters.data_getters import save_to_s3
+from dap_prinz_green_jobs import BUCKET_NAME
+
+import pandas as pd
+from argparse import ArgumentParser
+from tqdm import tqdm
+from sklearn.metrics.pairwise import cosine_similarity
+import numpy as np
+
+# instantiate GreenMeasures class here
+gm = GreenMeasures(config_name="base")
+
+if __name__ == "__main__":
+    parser = ArgumentParser()
+    parser.add_argument("--production", type=bool, default=True)
+
+    args = parser.parse_args()
+
+    production = args.production
+
+    # load and reformat relevant data
+    logger.info("loading and reformattting datasets...")
+
+    # step 0. load ojo related data - job title, skills and embeddings
+
+    ojo_job_title_raw = get_ojo_job_title_sample()
+    # ojo_skills_raw = (
+    #     get_ojo_skills_sample()
+    #     # drop nas in skill_label
+    #     .dropna(subset=["skill_label"])
+    # )
+    # # step 0.1 load embeddings (extracted skills and green skills taxonomy) and green skills taxonomy df
+    # extracted_skill_embeddings = get_extracted_skill_embeddings()
+    # extracted_skill_embeddings_skills = list(extracted_skill_embeddings.keys())
+
+    # green_taxonomy_embeddings = get_green_skills_taxonomy_embeddings()
+    # green_taxonomy_embeddings_skills = list(green_taxonomy_embeddings.keys())
+
+    # green_skills_taxonomy = get_green_skills_taxonomy()
+
+    # # reformat it to be a list of dictionaries for GreenMeasures
+    ojo_sample_raw = ojo_job_title_raw[["id", "job_title_raw", "company_raw"]].rename(
+        columns={
+            "job_title_raw": gm.job_title_key,
+            "company_raw": gm.company_name_key,
+        }
+    )
+    # we're not using descriptions so lets not load them to save time
+    ojo_sample_raw = list(ojo_sample_raw.T.to_dict().values())
+
+    # lets load the already mapped green skills from extract_green_skills.py instead of mapping them again
+    green_skills = get_mapped_green_skills()["SKILL MEASURES"]
```

Comment:
I think we have a bit of a funny and potentially prone-to-issues thing going on here. It seems this script can't be run with new job adverts as a one off, i.e. `extract_green_skills.py` **needs** to be run before `extract_green_measures.py` is run and not only that but we need to be certain that the job adverts inputted when `extract_green_skills.py` is run are the same as when `extract_green_measures.py` is run. At the moment we have nothing that tests that - `get_mapped_green_skills` is reading the job advert:skills from "outputs/data/green_skills/20230526_all_matched_skills.json" and this could have been created using a different dataset of job adverts than what this script might be processing?

It seems there is a bit of a long chain of things to run, from what I see it is:
1. `python dap_prinz_green_jobs/pipeline/green_measures/skills/skill_embedding.py --config_name "extract_green_skills_esco"`
2. `python dap_prinz_green_jobs/pipeline/ojo_application/extract_green_skills.py`
3. `python dap_prinz_green_jobs/pipeline/ojo_application/extract_green_measures.py`

I'd suggest this needs to be rewritten at some point (not for this PR though) so that we have one script that can input any set of job adverts and output the 3 measures. It could still read from "outputs/data/green_skill_lists/extracted_skills_embeddings.json" (and potentially even a skill:green flag) to help speed it up though, but the code should be prepared to calculate the embedding/match if the skill isn't found in this list.

Does that make sense or have I confused something?

Reply:
The skills code that's hashed out should replace 2. so the pipeline could be:

1. python dap_prinz_green_jobs/pipeline/green_measures/skills/skill_embedding.py --config_name "extract_green_skills_esco"
2. python dap_prinz_green_jobs/pipeline/ojo_application/extract_green_measures.py

but totally agreed that it should be re-written eventually once the skills side of things is more ironed out!